crates/core/src/config/secret.rs

```diff
@@ -0,0 +1,286 @@
+        }
+    }
+}
+
+impl Secrets {
+    #[inline]
+    pub fn nonce(&self) -> XNonce {
+        *XNonce::from_slice(&self.nonce)
+    }
+
+    #[inline]
+    pub fn cipher(&self) -> XChaCha20Poly1305 {
+        XChaCha20Poly1305::new(GenericArray::from_slice(&self.cipher))
+    }
+
+    #[inline]
+    pub fn transport_keypair(&self) -> &TransportKeypair {
+        &self.transport_keypair
+    }
+}
+
+fn read_nonce(path_to_nonce: impl AsRef<Path>) -> std::io::Result<[u8; NONCE_SIZE]> {
+    let path_to_nonce = path_to_nonce.as_ref();
+    let mut nonce_file = File::open(path_to_nonce).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to open key file {}: {e}", path_to_nonce.display()),
+        )
+    })?;
+    let mut buf = [0u8; NONCE_SIZE];
+    nonce_file.read_exact(&mut buf).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to read key file {}: {e}", path_to_nonce.display()),
+        )
+    })?;
+
+    Ok::<_, std::io::Error>(buf)
+}
+
+fn read_cipher(path_to_cipher: impl AsRef<Path>) -> std::io::Result<[u8; CIPHER_SIZE]> {
+    let path_to_cipher = path_to_cipher.as_ref();
+    let mut cipher_file = File::open(path_to_cipher).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to open key file {}: {e}", path_to_cipher.display()),
+        )
+    })?;
+    let mut buf = [0u8; CIPHER_SIZE];
+    cipher_file.read_exact(&mut buf).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to read key file {}: {e}", path_to_cipher.display()),
+        )
+    })?;
+
+    Ok::<_, std::io::Error>(buf)
+}
+
+fn read_transport_keypair(path_to_key: impl AsRef<Path>) -> std::io::Result<TransportKeypair> {
+    let path_to_key = path_to_key.as_ref();
+    let mut key_file = File::open(path_to_key).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to open key file {}: {e}", path_to_key.display()),
+        )
+    })?;
+    let mut buf = String::new();
+    key_file.read_to_string(&mut buf).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to read key file {}: {e}", path_to_key.display()),
+        )
+    })?;
+
+    let pk = rsa::RsaPrivateKey::from_pkcs1_pem(&buf).map_err(|e| {
+        std::io::Error::new(
+            std::io::ErrorKind::InvalidData,
+            format!("Failed to read key file {}: {e}", path_to_key.display()),
+        )
+    })?;
+
+    Ok::<_, std::io::Error>(TransportKeypair::from_private_key(pk))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_load() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let secrets_dir = temp_dir.path().join("secrets");
+        std::fs::create_dir_all(&secrets_dir).unwrap();
+
+        let transport_keypair = TransportKeypair::new();
+        let nonce = DelegateRequest::DEFAULT_NONCE;
```

Comment:
we want 2 test cases at least, one where we load default values, and an other where we load different values , and assert both cases work correctly